modules/system/classes/PluginManager.php

```diff
@@ -676,7 +676,10 @@ public function findMissingDependencies()
                 }
 
                 if (!in_array($require, $missing)) {
-                    $missing[] = $require;
+                    $missing[] = [
+                        'parent_code' => $this->getIdentifier($plugin),
+                        'code' => $require,
+                    ];
```

Comment:
The structure should be in the form of 

`['Parent.Plugin' => $arrayOfMissingDependencies]`, and should be documented as such in the docblock as well as changed in the report widget and the controller.